synapse/storage/receipts.py

```diff
@@ -95,6 +95,31 @@ def get_receipts_for_user(self, user_id, receipt_type):
         defer.returnValue({row["room_id"]: row["event_id"] for row in rows})
 
     @defer.inlineCallbacks
+    def get_receipts_for_user_with_orderings(self, user_id, receipt_type):
+        def f(txn):
+            sql = (
+                "SELECT rl.room_id, rl.event_id,"
+                " e.topological_ordering, e.stream_ordering"
+                " FROM receipts_linearized rl,"
+                " events e"
+                " WHERE rl.room_id = e.room_id"
```

Comment:
Generally I prefer to use explicit joins, as it makes it easier to see what the actual conditions are. i.e.:

```SQL
SELECT rl.room_id, rl.event_id, e.topological_ordering, e.stream_ordering FROM receipts_linearized AS rl
INNER JOIN events AS e USING (room_id, event_id)
WHERE user_id = ?
```